### PR TITLE
Point datatype

### PR DESCRIPTION
**Pull request by swilly22**
#1513

### CODE REVIEW

File: src/arithmetic/point_funcs/point_funcs.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		DegreeToRadians(p2.point.longitude) };
+
+	float dlat = lat[1] - lat[0];
+	float dlon = lon[1] - lon[0];
+
+	// a = sin²(Δφ/2) + cos φ1 ⋅ cos φ2 ⋅ sin²(Δλ/2)
+	float a = pow(sin(dlat / 2), 2) + cos(lat[0]) * cos(lat[1]) * pow(sin(dlon / 2), 2);
+
+	// c = 2 * atan2( √a, √(1−a) )
+	float c = 2 * atan2(sqrt(a), sqrt(1-a));
```
Comment (jeffreylovitz) [inline src/arithmetic/point_funcs/point_funcs.c:78]: This step is different than the one Wikipedia lists for the Haversine formula. Is it a different formula, or are they equivalent expressions?

**Comment by swilly22** [comment]
Introduce encoder/decoder V9 to deal with the point data-type.

File: src/resultset/formatters/resultset_replyverbose.c [hunk 2]
```diff
@@ -178,6 +182,15 @@ static void _ResultSet_VerboseReplyWithMap(RedisModuleCtx *ctx, SIValue map) {
 	rm_free(str);
 }
 
+static void _ResultSet_VerboseReplyWithPoint(RedisModuleCtx *ctx, SIValue point) {
+	// point({latitude:56.7, longitude:12.78})
+	char buffer[256];
+	int bytes_written = sprintf(buffer, "point({latitude:%f, longitude:%f})",
```
Comment (jeffreylovitz) [inline src/resultset/formatters/resultset_replyverbose.c:188]: Neo4j prints points using the format `{x: lat_val, y: lon_val}`. Would you prefer to be consistent here?

File: src/value.h [hunk 3]
```diff
@@ -84,6 +85,10 @@ typedef struct SIValue {
 		void *ptrval;
 		struct Pair *map;
 		struct SIValue *array;
+		struct {
+			float latitude;   // 32 bit
+			float longitude;  // 32 bit
```
Comment (jeffreylovitz) [inline src/value.h:90]: I understand the desire for a struct size of 8, but are floats sufficiently precise for lat/lon values? I saw a lot of floating point error with some minor testing.

If this is a significant problem, we should switch to a heap-allocated struct.

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]
This PR requires documentation updates and investigation into the memory error on indexing. Newly-added tests will fail until the merge of https://github.com/RedisGraph/redisgraph-py/pull/121

**Pull request merged**